tests/Resources/ti.ui.tableview.addontest.js

```diff
@@ -0,0 +1,35 @@
+/*
+ * Appcelerator Titanium Mobile
+ * Copyright (c) 2015-Present by Appcelerator, Inc. All Rights Reserved.
+ * Licensed under the terms of the Apache Public License
+ * Please see the LICENSE included with this distribution for details.
+ */
+/* eslint-env mocha */
+/* global Ti */
+/* eslint no-unused-expressions: "off" */
+'use strict';
+var should = require('./utilities/assertions');
+
+describe('Titanium.UI.TableView', function () {
+
+	it.android('SearchView persistence', function (finish) {
+		var	tableData = [ { title: 'Apples' }, { title: 'Bananas' }, { title: 'Carrots' }, { title: 'Potatoes' } ],
+			searchView = Ti.UI.Android.createSearchView(),
+			table = Ti.UI.createTableView({
+				height: '80%',
+				search: searchView,
+				data: tableData
+			}),
+			win = Ti.UI.createWindow();
+		function removeAndAddTable() {
+			table.removeEventListener('postlayout', removeAndAddTable);
+			win.remove(table);
+			win.add(table);
+			finish();
```

Comment:
When a function is used as a callback/event listener it will execute "async" and can behave a little strange under unit tests if there's an Error thrown. Typically you'd want to wrap the body of the function where you're doing the work in a `try` block and place the `finish()` call last inside the block - with a `catch(err)` block that calls `finish(err);` to handle error cases.

i.e.
```javascript
function removeAndAddTable() {
  try {
 	table.removeEventListener('postlayout', removeAndAddTable);
 	win.remove(table);
 	win.add(table);
 	finish();
  } catch (err) {
    finish(err);
  }
}
```

Without that I've seen different platforms behave differently - Windows sometimes crashes, the other sometimes just time out the test, leading you to think that there was a timeout/hang (when there wasn't).